### PR TITLE
Support this.prop = expr; assignments as declarations for ES6 JS classes

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
@billti noticed this didn't actually make it in to the original JSDoc branch as planned

### CODE REVIEW

File: tests/cases/fourslash/javaScriptClass1.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+///<reference path="fourslash.ts" />
+
+// Classes have their shape inferred from assignments
+// to properties of 'this' in the constructor
+
+// @allowNonTsExtensions: true
+// @Filename: Foo.js
+//// class Foo {
+////    constructor() {
+////        this.bar = 'world';
```
Comment (billti) [inline tests/cases/fourslash/javaScriptClass1.ts]: Can we add a method assignment also for completeness?

File: tests/cases/fourslash/javaScriptClass1.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// Classes have their shape inferred from assignments
+// to properties of 'this' in the constructor
+
+// @allowNonTsExtensions: true
+// @Filename: Foo.js
+//// class Foo {
+////    constructor() {
+////        this.bar = 'world';
+////        this.thing = 42;
+////        this.union = 'foo';
```
Comment (billti) [inline tests/cases/fourslash/javaScriptClass1.ts]: Can we check if JsDoc works for the assignment member, e.g. `@type`, or even `@params` if the assignment is a method?

File: tests/cases/fourslash/javaScriptClass1.ts (hunk 1 above)
Comment (RyanCavanaugh) [inline tests/cases/fourslash/javaScriptClass1.ts]: Done

File: tests/cases/fourslash/javaScriptClass1.ts (hunk 2 above)
Comment (RyanCavanaugh) [inline tests/cases/fourslash/javaScriptClass1.ts]: Added test for `@type`

**Comment by billti** [comment]
:+1:

**Comment by mhegazy** [comment]
@RyanCavanaugh can we close this?

**Pull request closed without merging**